lib/v4/MessageBuilder.ts

```diff
@@ -0,0 +1,75 @@
+import { type NonEmptyArray } from '../utils/NonEmptyArray.ts';
+import { defaultErrorMapOptions } from './errorMap/index.ts';
+import type * as zod from 'zod/v4/core';
+
+export type ZodIssue = zod.$ZodIssue;
+
+export type MessageBuilder = (issues: NonEmptyArray<ZodIssue>) => string;
+
+const stubErrorMap: zod.$ZodErrorMap<zod.$ZodIssue> = (issue) => {
+  return issue.message;
+};
+
+export type MessageBuilderOptions = {
+  prefix: string | null | undefined;
+  prefixSeparator: string;
+  maxIssuesInMessage: number;
+  issueSeparator: string;
+  error: zod.$ZodErrorMap<zod.$ZodIssue>;
```

Comment:
dev: `MessageBuilder` doesn't do any message formatting on its own but it can accept a custom `errorMap` as option, or use the default `stubErrorMap` which just returns any message provided by zod.